jax/_src/scipy/linalg.py

```diff
@@ -2243,3 +2243,51 @@ def _binom(n, k):
   b = lax.lgamma(n - k + 1.0)
   c = lax.lgamma(k + 1.0)
   return lax.exp(a - b - c)
+
+
+@jit
+def solve_sylvester(A: ArrayLike, B: ArrayLike, C: ArrayLike) -> Array:
+    """
+    Solves the Sylvester equation:
+    .. math::
+
+      AX + XB = C
+
+    Args:
+      A: Matrix of shape m x m
+      B: Matrix of shape n x n
+      C: Matrix of shape m x n
+
+    Returns:
+      X: Matrix of shape m x n
+
+    Examples:
+      >>> A = jax.numpy.array([[1, 2], [3, 4]])
+      >>> B = jax.numpy.array([[5, 6], [7, 8]])
+      >>> C = jax.numpy.array([[6, 8], [10, 12]])
+      >>> jax.scipy.linalg.solve_sylvester(A, B, C)
+      Array([[1.00000000e+00, 2.36169983e-16],
```

Comment:
This doctest is failing, because the exact rounding of values near zero differs between platforms. Typically we address this by adjusting the print options; for example: https://github.com/jax-ml/jax/blob/2c838d4ae3fff34242d5a2993e6892c8214c9177/jax/_src/scipy/linalg.py#L805-L813